test/src/utils.dart

```diff
@@ -39,53 +32,43 @@ void delete(Directory dir) {
 }
 
 void init() {
-  ResourceProvider resourceProvider = PhysicalResourceProvider.INSTANCE;
-  sdkDir = new FolderBasedDartSdk(
-      resourceProvider, resourceProvider.getFolder(getSdkDir().path));
-
+  sdkDir = getSdkDir();
+  sdkPackageMeta = new PackageMeta.fromSdk(sdkDir);
   setConfig();
 
-  analyzerHelper = new AnalyzerHelper();
-  var pathsForTestLib = [
-    'lib/example.dart',
-    'lib/two_exports.dart',
-    'lib/fake.dart',
-    'lib/anonymous_library.dart',
-    'lib/another_anonymous_lib.dart',
-    'lib/is_deprecated.dart',
-    'lib/reexport_one.dart',
-    'lib/reexport_two.dart',
-  ];
-
-  testPackage = _bootPackage(pathsForTestLib, 'testing/test_package', false);
-  testPackageGinormous =
-      _bootPackage(pathsForTestLib, 'testing/test_package', true);
+  testPackage = bootBasicPackage(
+      'testing/test_package', ['css', 'code_in_comments', 'excluded'], false);
+  testPackageGinormous = bootBasicPackage(
+      'testing/test_package', ['css', 'code_in_commnets', 'excluded'], true);
 
-  testPackageSmall =
-      _bootPackage(['lib/main.dart'], 'testing/test_package_small', false);
+  testPackageSmall = bootBasicPackage('testing/test_package_small', [], false);
+  testPackageSdk = bootSdkPackage();
 }
 
-Package _bootPackage(Iterable<String> libPaths, String dirPath,
-    bool withAutoIncludedDependencies) {
-  String fullDirPath = p.join(Directory.current.path, dirPath);
-  Iterable<LibraryElement> libElements = libPaths.map((libFile) {
-    Source source = analyzerHelper.addSource(p.join(fullDirPath, libFile));
-    return analyzerHelper.resolve(source);
-  });
-
-  if (withAutoIncludedDependencies) {
-    return Package.withAutoIncludedDependencies(
-        libElements,
-        new PackageMeta.fromDir(new Directory(dirPath)),
-        new PackageWarningOptions());
-  } else {
-    return new Package(
-        libElements,
-        new PackageMeta.fromDir(new Directory(dirPath)),
-        new PackageWarningOptions());
-  }
+Package bootSdkPackage() {
+  Directory dir = new Directory(p.current);
+  return new PackageBuilder(
+          dir, [], [], sdkDir, sdkPackageMeta, [], [], true, false)
+      .buildPackage();
 }
 
+Package bootBasicPackage(
+    String dirPath, List<String> excludes, bool withAutoIncludedDependencies) {
+  Directory dir = new Directory(dirPath);
+  return new PackageBuilder(
+          dir,
+          excludes,
+          [],
+          sdkDir,
+          new PackageMeta.fromDir(new Directory(dirPath)),
+          [],
+          [],
+          true,
+          withAutoIncludedDependencies)
```

Comment:
a trailing comma here (and some other places with less than ideal formatting) may make dartfmt format this better

Reply:
(I find that trailing commas can make things this split lines dartfmt better)